### DISCUSSION TITLE
Ability to save/export diff editor content as formatted html or png

### DISCUSSION THREAD

**Discussion by kvarforl** [microsoft/monaco-editor#3653] [Q&A]
**Context:**
For the project we are using the diff editor in, users will often need to create diffs between two sets of code for code reviews.
These diffs are currently created and saved using Beyond Compare so that the diff file can be stored as documentation.
It's frustrating to not be able to easily export the formatted diff content directly from the editor, and instead have to copy paste into a separate app.

**Feature Request:**
-  An easy way to access all of the content of the diff editor, including formatting, and content beyond what's in the viewport.
-  Better yet, a method that will export the formatted diff to a file.

**Alternatives Considered:**
We've tried using libraries like html2canvas and other methods grabbing the html of the DiffEditor component, but both of these only capture what is currently in the viewport. If either side of the diff requires scrolling to see the whole content, that content won't be displayed in the capture.
Any other ideas or help would be appreciated!

**Comment by ianmcgowan** [comment]
A workaround, but a Beyond Compare vscode plugin was released by Scooter Software in Jan 2024 and it makes the process a little quicker.  I too have the requirement to upload an html diff into a ticketing system, and the plugin makes it more efficient.

[https://marketplace.visualstudio.com/items?itemName=ScooterSoftware.bcompare-vscode](https://marketplace.visualstudio.com/items?itemName=ScooterSoftware.bcompare-vscode)
